jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostDFSuite.scala

```diff
@@ -138,6 +141,27 @@ class XGBoostDFSuite extends FunSuite with PerTest with TableDrivenPropertyCheck
     assert(xgbEstimatorCopy.fromParamsToXGBParamMap("objective").toString === "binary:logistic")
   }
 
+  test("test checkpointing with dataframe-based training") {
```

Comment:
would you update updateCheckpoint() method in CheckPointManager class with a switch controlling if it clears the previous version of checkpoint 

and then you disable checkpoint cleaning in the test, and count the files and check file names in this test?

Reply:
As we do not have access to the CheckPointManager directly because it is instantiated inside the estimator, this would require to add a new xgboost parameter to switch off clearing previous checkpoint versions. I think it is a good idea as that is an actual feature we might want to use but I feel it would be cleaner to put that into another pull-request as this only intend to fix a typo in the parameter name. What do you think @CodingCat ?

Reply:
I think it's fine to put this into the same PR, as the major point for this feature is to validate the correctness of checkpoint_interval

Reply:
ping